docs/source/developer_guide/architecture.md

```diff
@@ -80,38 +63,20 @@ The stage identifier is a unique string used in both logging and creating the st
 
 ### Type Inference
 
-To perform work, each stage needs to know what type of data it will be operating on. Since Morpheus can pass any type of
-data from stage to stage, the pipeline must ensure compatible types at every edge connection between stages. This
-process is called stage type inference and is performed during the pipeline build phase.
+To perform work, each stage needs to know what type of data it will be operating on. Since Morpheus can pass any type of data from stage to stage, the pipeline must ensure compatible types at every edge connection between stages. This process is called stage type inference and is performed during the pipeline build phase.
```

Comment:
nit: s/will be operating/operates/  # Present tense is best. Nearly all conjugations of to be can be avoided.

nit: s/Since/Because/ or omit.  # Because is one word with one meaning.

There's some anthropomorphism here too--suggesting that a stage has the human-like ability to know.  I'm afraid that I don't know enough about Morpheus to suggest a good alternative.

Reply:
I'm going to keep the future tense of `will be operating` since the paragraph is discussing a pre-flight check that happens during the pipeline build.